server/plugins/renderer/renderer.module.spec.js

```diff
@@ -253,9 +303,50 @@ describe('Renderer Plugin', () => {
                 'stencil-options': '{"get_template_file":true,"get_data_only":true}',
                 'stencil-version': PACKAGE_INFO.config.stencil_version,
                 'accept-encoding': 'identity',
+                'stencil-custom-templates':
+                    '{"brand":{"a":"/abc/"},"category":{},"page":{},"product":{}}',
             });
         });
 
+        it('should send a request to the storefront server with added custom layouts to headers', async () => {
+            const limitedServerOptions = {
+                dotStencilFile: {
+                    storeUrl,
+                    normalStoreUrl,
+                    port: 4002,
+                    username: 'testUser',
+                    token: '6832b1c755bb9de13aa8990216a69a7623043fd7',
+                },
+                useCache: false,
+                themePath: themeConfigManager.themePath,
+            };
+
+            const newServer = await Server.create(limitedServerOptions);
+
+            // Don't log errors during the test
+            newServer.ext('onPostHandler', (request, h) => {
+                if (request.response.isBoom) {
+                    return h.response().code(500);
+                }
+                return h.continue;
+            });
+
+            axiosMock.onGet().reply(200, {});
+            await newServer.inject(browserRequest);
+
+            expect(axiosMock.history.get[0].headers).toMatchObject({
+                cookie: browserRequest.headers.cookie,
+                host: new URL(storeUrl).host,
+                'stencil-cli': PACKAGE_INFO.version,
+                'stencil-options': '{"get_template_file":true,"get_data_only":true}',
+                'stencil-version': PACKAGE_INFO.config.stencil_version,
+                'accept-encoding': 'identity',
+                'stencil-custom-templates': '{"brand":{},"category":{},"page":{},"product":{}}',
```

Comment:
here we receive `'{"brand":{"a":"/abc/"},"category":{},"page":{},"product":{}}'` instead of `'{"brand":{},"category":{},"page":{},"product":{}}'`